cmd/bnsd/x/blueaccount/codec.proto

```diff
@@ -0,0 +1,192 @@
+}
+
+// BlockchainAddress represents a blochain address. This structure clubs together
+// blokchain ID together with an address on that network. It is used to point
+// to an address on any blockchain network.
+message BlockchainAddress {
+  // An arbitrary blockchain ID.
+  string blockchain_id = 1 [(gogoproto.customname) = "BlockchainID"];
+  // An address on the specified blockchain network. Address is not a
+  // weave.Address as we cannot know what is the format of an address on the
+  // chain that this token instance links to. Because we do not know the rules
+  // to validate an address for any blockchain ID, this is an arbitrary bulk of
+  // data.
+  // It is more convinient to always use encoded representation of each address
+  // and store it as a string. Using bytes while compact is not as comfortable
+  // to use.
+  string address = 2;
+}
+
+// Configuration is a dynamic configuration used by this extension, managed by
+// the functionality provided by gconf package.
+message Configuration {
+  weave.Metadata metadata = 1;
+  // Owner is present to implement gconf.OwnedConfig interface
+  // This defines the Address that is allowed to update the Configuration object and is
+  // needed to make use of gconf.NewUpdateConfigurationHandler
+  bytes owner = 2 [(gogoproto.casttype) = "github.com/iov-one/weave.Address"];
+  // Valid domain defines a regular expression that every valid domain of an
+  // account must match. Rule must not include the asterisk (*) character.
+  string valid_domain = 3;
+  // Valid name defines a regular expression that every valid name of an
+  // account must match. Rule must not include the asterisk (*) character.
+  string valid_name = 4;
+  // Domain review defines the duration of the domain renewal period.
+  int64 domain_renew = 6 [(gogoproto.casttype) = "github.com/iov-one/weave.UnixDuration"];
+}
+
+// UpdateConfigurationMsg is used by the gconf extension to update the
+// configuration.
+message UpdateConfigurationMsg {
+  weave.Metadata metadata = 1;
+  Configuration patch = 2;
+}
+
+// RegisterDomainMsg is issuing registration of a new domain. Requested
+// domain name must be available (not yet registed) in order for the request to
+// succeed.
+// Registering a domain creates an account with no name that belongs to this
+// domain.
+message RegisterDomainMsg {
+  weave.Metadata metadata = 1;
+  // Owner holds the address of the newly registered domain. If not provided,
+  // main signer is used.
+  bytes owner = 2 [(gogoproto.casttype) = "github.com/iov-one/weave.Address"];
+  // Domain part of the account that should be registered.
+  string domain = 3;
+  // Client token is an arbitrary data that can be provided by a middleman
+  // (i.e. retailer) that although does not participate directly in the
+  // transaction, helped a user to register a domain. Storing a client token
+  // helps identify contribution of such party later.
+  bytes client_token = 4;
```

Comment:
I am not sure `client_token` is the best name 🤔

Reply:
I did not want to use word "tracking" as this often has a negative meaning.

What other name could I use?

Reply:
`third_party_token` ?